### PR TITLE
Metron 145: Automate Creation of Vagrant Images

### PR DESCRIPTION
**Pull request by dlyle65535**

### CODE REVIEW

**Comment by dlyle65535** [comment]
Looks like I got got buy the NPE in integration testing. This passed in my repo, I'll close and re-open.

**Comment by nickwallen** [comment]
@dlyle65535 How can I test this?  Looking forward to seeing it in action.

**Comment by dlyle65535** [comment]
I'm hopeful that the README has sufficient steps. Please let me know if I'm wrong about that and I'll correct it. Thanks!

File: metron-deployment/packer-build/ansible.cfg [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+host_key_checking = false
+library = ../extra_modules
+roles_path = ../roles
+pipelining = True
+remote_user = vagrant
+log_path = ./ansible.log
+
+# fix for "ssh throws 'unix domain socket too long' " problem
+[ssh_connection]
+control_path = %(directory)s/%%h-%%p-%%r
```
Comment (nickwallen) [inline metron-deployment/packer-build/ansible.cfg]: Should we follow the lead of PR #122 here?

File: metron-deployment/packer-build/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Prerequisites
+---------------------
+- Packer 0.10.1 https://www.packer.io/
+- Virtualbox 5.0.16 https://www.virtualbox.org/
+
+Build Both Images
+---------------------- 
+  Navigate to <your-project-directory>/metron-deployment/packer-build
+  Execute bin/bento
```
Comment (nickwallen) [inline metron-deployment/packer-build/README.md]: Seems like I need to pass an argument to `bin/bento`?  Maybe build?  Should update the README.

```
nallen @ ~/tmp/incubator-metron-pr-123/metron-deployment/packer-build
$ ./bin/bento
Usage: bento [SUBCOMMAND [options]]

build     :   build one or more templates
help      :   prints this help message
list      :   list all templates in project
normalize :   normalize one or more templates
```

Comment (nickwallen) [inline metron-deployment/packer-build/README.md]: If I use `build`, I seem to run into an error.  What am I doing wrong?

```
nallen @ ~/tmp/incubator-metron-pr-123/metron-deployment/packer-build
$ ./bin/bento build
==> Starting build for templates: ["hdp-centos-6.7", "metron-centos-6.7"]
==> [hdp-centos-6.7] Building: 'packer build -var-file=/var/folders/hb/dp9xm3q92gq3w9w9btb0kqyw0000gn/T/hdp-centos-6.7-metadata-var-file20160518-26378-12zwyn0 hdp-centos-6.7.json'
>>> [hdp-centos-6.7] Error building, exited pid 26387 exit 127
```

File: metron-deployment/packer-build/ansible.cfg (hunk 1 above)
Comment (dlyle65535) [inline metron-deployment/packer-build/ansible.cfg]: Good call, I think so.

File: metron-deployment/packer-build/README.md (hunk 2 above)
Comment (dlyle65535) [inline metron-deployment/packer-build/README.md]: Any chance you don't have packer installed?

**Comment by nickwallen** [comment]
I was a good 20-40 minutes through the process when it errored out because I had not packaged the Metron platform code.  Might be good to call this out in the README as a preliminary step.

**Comment by nickwallen** [comment]
It seems to be working great so far though.  Last night,I had not installed Packer (doh) which was causing that issue.  I'm so used to relying on Ansible to just install everything for me. :)

**Comment by nickwallen** [comment]
👍

**Comment by cestella** [comment]
+1

**Comment by merrimanr** [comment]
+1

**Pull request closed without merging**